projects/wave-core-new/src/lib/datatable/table/full-display/full-display.component.ts

```diff
@@ -0,0 +1,63 @@
+import {Component, OnInit, ChangeDetectionStrategy, Inject} from '@angular/core';
+import {MAT_DIALOG_DATA} from '@angular/material/dialog';
+import {Clipboard} from '@angular/cdk/clipboard';
+import {Geometry, MultiPolygon} from 'ol/geom';
+
+/**
+ * Opened as modal dialog to display a full set of coordinates and allow copying to clipboard
+ */
+@Component({
+    selector: 'wave-full-display',
+    templateUrl: './full-display.component.html',
+    styleUrls: ['./full-display.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class FullDisplayComponent implements OnInit {
+    xCoords: string[] = [];
+    yCoords: string[] = [];
+
+    constructor(
+        @Inject(MAT_DIALOG_DATA) public data: {xStrings: string[]; yStrings: string[]; geometry: Geometry},
+        private clipboard: Clipboard,
+    ) {}
+
+    ngOnInit(): void {
+        this.xCoords = this.data.xStrings;
+        this.yCoords = this.data.yStrings;
+    }
+
+    copyToClipboard(): void {
+        const type = this.data.geometry.getType();
+        const prettyCoords = JSON.stringify((this.data.geometry as MultiPolygon).getCoordinates(), null, 2);
+        const result = `{ "type": "${type}",\n  "coordinates":\n    ${prettyCoords}\n}`;
+        const pending = this.clipboard.beginCopy(result);
+        let remainingAttempts = 3;
+        function attempt(): void {
+            const copied = pending.copy();
+            if (!copied && --remainingAttempts) {
+                setTimeout(attempt);
+            } else {
+                pending.destroy();
+            }
+        }
+        attempt();
```

Comment:
Please refactor this to be one method that encapsulates this loop and callback stuff. You do exactly the same in the other method.

Reply:
Refactored to a method c5fa8fea182296357d055b3a755854320a526ab7